From dfc129a9f7a6c303b45ecf646a252d7e4f3d4942 Mon Sep 17 00:00:00 2001
From: Vihang Karajgaonkar <vihang@cloudera.com>
Date: Wed, 25 May 2016 16:28:22 -0500
Subject: [PATCH 0610/1164] CDH-32902 : HIVE-13797: Provide a connection
 string example in beeline (Vihang Karajgaonkar,
 reviewed Lefty Leverenz and Sergio Pena)

Change-Id: I934f95bbde3247f11c9d538753bde8c69cd6ca4d
---
 beeline/src/main/resources/BeeLine.properties |   16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/beeline/src/main/resources/BeeLine.properties b/beeline/src/main/resources/BeeLine.properties
index 2e987fd..ad7280d 100644
--- a/beeline/src/main/resources/BeeLine.properties
+++ b/beeline/src/main/resources/BeeLine.properties
@@ -174,6 +174,16 @@ cmd-usage: Usage: java org.apache.hive.cli.beeline.BeeLine \n \
 \  --delimiterForDSV=DELIMITER     specify the delimiter for delimiter-separated values output format (default: |)\n \
 \  --isolation=LEVEL               set the transaction isolation level\n \
 \  --nullemptystring=[true/false]  set to true to get historic behavior of printing null as empty string\n \
-\  --help                          display this message
-
-
+\  --help                          display this message\n \
+\n \
+\  Example:\n \
+\   1. Connect using simple authentication to HiveServer2 on localhost:10000\n \
+\   $ beeline -u jdbc:hive2://localhost:10000 username password\n\n \
+\   2. Connect using simple authentication to HiveServer2 on hs.local:10000 using -n for username and -p for password\n \
+\   $ beeline -n username -p password -u jdbc:hive2://hs2.local:10012\n\n \
+\   3. Connect using Kerberos authentication with hive/localhost@mydomain.com as HiveServer2 principal\n \
+\   $ beeline -u "jdbc:hive2://hs2.local:10013/default;principal=hive/localhost@mydomain.com\n\n \
+\   4. Connect using SSL connection to HiveServer2 on localhost at 10000\n \
+\   $ beeline jdbc:hive2://localhost:10000/default;ssl=true;sslTrustStore=/usr/local/truststore;trustStorePassword=mytruststorepassword\n\n \
+\   5. Connect using LDAP authentication\n \
+\   $ beeline -u jdbc:hive2://hs2.local:10013/default <ldap-username> <ldap-password>\n \
-- 
1.7.9.5

